docs/modules/ROOT/pages/udc.adoc

```diff
@@ -1,226 +1,144 @@
+:deploy-syscall: xref:architecture_and_concepts:Smart_Contracts/system-calls-cairo1.adoc#deploy[deploy syscall]
+:deployer-interface: link:https://community.starknet.io/t/snip-deployer-contract-interface/2772[standard deployer interface]
+:interface-proposal: link:https://community.starknet.io/t/snip-deployer-contract-interface/2772[proposal for the standard deployer interface]
+:udc-proposal: link:https://community.starknet.io/t/universal-deployer-contract-proposal/1864[Universal Deployer Contract proposal]
+:previous-udc-docs: link:https://docs.openzeppelin.com/contracts-cairo/0.6.1/udc[previous UDC docs]
+:deploy_contract: xref:/api/udc.adoc#UniversalDeployer-deploy_contract[deploy_contract]
+:contract-deployed: xref:/api/udc.adoc#IUniversalDeployer-ContractDeployed[ContractDeployed]
 
-And since StarkNet has no deployment transaction type, it offers a canonical way to deploy smart contracts by following the https://community.starknet.io/t/snip-deployer-contract-interface/2772[standard deployer interface] and emitting a `ContractDeployed` event.
+The Universal Deployer Contract (UDC) is a singleton smart contract that wraps the {deploy-syscall} to expose it to any contract that doesn't implement it, such as account contracts. You can think of it as a standardized generic factory for Starknet contracts.
 
-TIP: For information on design decisions, see the https://community.starknet.io/t/universal-deployer-contract-proposal/1864[Universal Deployer Contract Proposal].
+And since Starknet has no deployment transaction type, it offers a standardized way to deploy smart contracts by following the {deployer-interface} and emitting a {contract-deployed} event.
 
-NOTE: The UDC address is `0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02bf` in mainnet, testnets, and starknet-devnet. This address may change in the future.
+For more information see the {interface-proposal}.
+For details on the motivation and the decision making process, see the {udc-proposal}.
 
-== Table of Contents
-* <<quickstart, Quickstart>>
-* <<deploying_the_udc,Deploying the UDC>>
-* <<usage,Usage>>
- ** <<deployer_dependent,Deployer dependent>>
- ** <<deployer_independent,Deployer independent>>
-* <<calculating_counterfactual_addresses,Calculating counterfactual addresses>>
-* <<api_specification,API specification>>
+== UDC contract address
 
-== Quickstart
+The UDC address is deployed at address `0xFIX-ME` in ADD-NETWORKS-WHEN-DEPLOYED.
 
-To deploy a contract first you need to make sure it's been declared and then send a `deployContract` transaction to the Universal Deployer Contract.
-Here's an example implementation in Python:
+NOTE: The first iteration of the UDC is deployed at address `0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02bf` in Mainnet, Goerli testnet, Sepolia testnet, and starknet-devnet. See the {previous-udc-docs} for more information.
```

Comment:
This NOTE should be either removed or rewritten. We should remove Goerli's mention, and probably add the inconsistency in the event (unique vs from_zero) between the two versions

Reply:
Removed!